extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java

```diff
@@ -84,179 +96,335 @@ FeatureBuildItem registerFeature() {
-                            //but will switch to Dependent if the is a Scope annotation with the value of prototype
-                            DotName cdiAnnotation = CDI_SINGLETON_ANNOTATION;
-                            if (classInfo.annotations().containsKey(SPRING_SCOPE_ANNOTATION)) {
-                                final AnnotationInstance annotation = classInfo.classAnnotation(SPRING_SCOPE_ANNOTATION);
-                                final AnnotationValue springScopeAnnotationValue = annotation.value();
-                                if (springScopeAnnotationValue != null &&
-                                        "prototype".equals(springScopeAnnotationValue.asString())) {
-                                    cdiAnnotation = CDI_DEPENDENT_ANNOTATION;
-                                }
-                            }
-                            annotationsToAdd.add(create(
-                                    cdiAnnotation,
-                                    context.getTarget(),
-                                    new ArrayList<>()));
-
-                            //check if the spring annotation defines a name for the bean
-                            final AnnotationValue beanNameAnnotationValue = classInfo
-                                    .classAnnotation(springStereotypeAnnotation).value();
-                            addCDINamedAnnotation(context, beanNameAnnotationValue, annotationsToAdd);
-
-                            break;
-                        }
-                    }
+    private DotName getScope(final ClassInfo clazz) {
+        if (!clazz.annotations().containsKey(SPRING_SCOPE_ANNOTATION)) {
+            return null;
+        }
+        final AnnotationValue value = clazz.classAnnotation(SPRING_SCOPE_ANNOTATION).value();
+        if (value != null) {
+            switch (value.asString()) {
+                case "singleton":
+                    return CDI_SINGLETON_ANNOTATION;
+                case "request":
+                    return CDI_REQUEST_SCOPED_ANNOTATION;
+                case "global session":
+                case "application":
+                    return CDI_APP_SCOPED_ANNOTATION;
+                case "session":
+                    return CDI_SESSION_SCOPED_ANNOTATION;
+                case "websocket":
+                case "prototype":
+                    return CDI_DEPENDENT_ANNOTATION;
+            }
+        }
+        return null;
+    }
 
-                    if (classInfo.annotations().containsKey(CONFIGURATION_ANNOTATION)) {
-                        annotationsToAdd.add(create(
-                                CDI_APP_SCOPED_ANNOTATION,
-                                context.getTarget(),
-                                new ArrayList<>()));
+    /**
+     * @return All the annotations in the index ordered with dependents after their dependencies
+     */
+    private List<ClassInfo> getOrderedAnnotations(final IndexView index) {
```

Comment:
:+1: 

Can you please update the comments to explain *why* the dependents needs to be after the dependencies so future readers of the code don't have to figure that out on the consumer side?

Reply:
Done